pkg/kubelet/gpu/nvidia/nvidia_gpu_manager_test.go

```diff
@@ -36,6 +38,72 @@ func (tapl *testActivePodsLister) GetActivePods() []*v1.Pod {
 	return tapl.activePods
 }
 
+func makeTestProcFile(t *testing.T, content []byte) string {
+	tmpfile, err := ioutil.TempFile("", "tmp")
+	if err != nil {
+		t.Fatal(err)
+	}
+	if _, err := tmpfile.Write(content); err != nil {
+		t.Fatal(err)
+	}
+	if err := tmpfile.Close(); err != nil {
+		t.Fatal(err)
+	}
+	devicesFile := tmpfile.Name()
+	return devicesFile
+}
+
+func makeDummyDevDir(t *testing.T, fakeDevices []string) string {
+	content := []byte("dummy content")
+	dir, err := ioutil.TempDir("", "tmpdir")
+	if err != nil {
+		t.Fatal(err)
+	}
+	for _, fileName := range fakeDevices {
+		tmpfn := filepath.Join(dir, fileName)
+		if err := ioutil.WriteFile(tmpfn, content, 0666); err != nil {
+			t.Fatal(err)
+		}
+	}
+	return dir
+}
+
+func TestValidGPUS(t *testing.T) {
+	testProc := makeTestProcFile(t, []byte("aaa\t10debbb\tccc\nddd\teee\tfff\nggg\t10dehhh\tiii\n"))
+	testDir := makeDummyDevDir(t, []string{"nvidia1", "nvidia2"})
+	foundGPUs, err := expectedGPUs(testProc)
+	as := assert.New(t)
+	as.Nil(err)
+	as.Equal(foundGPUs, true)
+	testGpuManager := &nvidiaGPUManager{
+		allGPUs: sets.NewString(),
+	}
+	err = testGpuManager.discoverGPUs(testProc, testDir)
+	as.Nil(err)
+}
+
+func TestNoGPUS(t *testing.T) {
+	testProc := makeTestProcFile(t, []byte("aaa\tbbbb\tcccc\ndddd\teeee\tffff\ngggg\thhhh\tiiii\n"))
+	noFoundGPUs, err := expectedGPUs(testProc)
+	as := assert.New(t)
+	as.Nil(err)
+	as.Equal(noFoundGPUs, false)
```

Comment:
Also test that `discoverGPUs` doesn't return an error in this case. Would have caught the bug I pointed out above.